RHDEVDOCS-6064: Sizing requirements for argocd-redis (#77736)
openshift/openshift-docs: modules/sizing-requirements-for-gitops.adoc

----
oc edit argocd <name of argo cd> -n namespace
----

[id="sizing-requirements-for-argocd-redis_{context}"]
== Sizing requirements for argocd-redis

> During the capacity planning stage for your application in the {gitops-title} Operator, you must ensure that an adequate amount of resources, such as memory, CPU, and storage, is allocated for the `argocd-redis` pod.

Reviewer: ```suggestion
During the capacity planning stage for your application in the {gitops-title} Operator, you must ensure that an adequate amount of resources, such as memory, CPU, and storage, are allocated for the `argocd-redis` pod.
```

Author: Fixed.

The default memory limit for the Redis pod might not be enough to manage a large number of resources. In these instances, you must increase the memory limit, monitor the memory metrics, and change the memory configuration while the application deployment scales up.

The following command shows the example of the memory configuration for a Redis pod in the `openshift-gitops` namespace:
[source,terminal]
----

> oc get argocd -n openshift-gitops openshift-gitops -o json | jq '.spec.redis.resources'

Reviewer 2: ```suggestion
$ oc get argocd -n openshift-gitops openshift-gitops -o json | jq '.spec.redis.resources'
```
We use the pro

Author: Fixed.

----

.Example Output
[source,terminal]
----
{
    "limits": { #<1>
        "cpu": "500m",
        "memory": "256Mi"
  },
  "requests": { #<2>
    "cpu": "250m",
    "memory": "128Mi"
  }
}
----
<1> The highest resource limit threshold allocated to the pod.
<2> The lowest resource limit threshold allocated to the pod.

> The following example command changes the memory configuration for a Redis pod.  The highest resource limit threshold is set to 8 GiB and the lowest is set to 256 MiB.

Reviewer: ```suggestion
The following example command changes the memory configuration for a Redis pod. The highest resource limit threshold is set to 8 GiB and the lowest is set to 256 MiB.
```

Author: Fixed.

[source,terminal]
----

> oc patch argocd -n openshift-gitops openshift-gitops --type json -p '[{"op": "replace", "path": "/spec/redis/resources/limits/memory", "value": "8Gi"}, {"op": "replace", "path": "/spec/redis/resources/requests/memory", "value": "256Mi"}]'

Reviewer 2: ```suggestion
$ oc patch argocd -n openshift-gitops openshift-gitops --type json -p '[{"op": "replace", "path":  \
  "/spec/redis/resources/limits/memory", "value": "8Gi"}, {"op": "replace", "path": \
  "/spec/redis/resources/requests/memory", "value": "256Mi"}]' \
```
See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#backslash-for-long-lines-of-code

Author: Fixed.

> ----

Reviewer 3: This is a comment about the table in the "" section. Based on IBM unit guide, we should refer units differently: https://www.ibm.com/docs/en/ibm-style?topic=measurement-units#defining-abbreviations
I would suggest making the change to 1024 MiB instead of 1024M, etc.

Author: This comment will be addressed by the following PR:

https://github.com/openshift/openshift-docs/pull/78989

Hence, I am marking this comment as resolved.

oc edit argocd <name of argo cd> -n namespace
----

[id="sizing-requirements-for-argocd-redis_{context}"]
== Sizing requirements for argocd-redis

> During the capacity planning stage for your application in the {gitops-title} Operator, you must ensure that an adequate amount of resources such as memory, CPU, and storage are allocated for the `argocd-redis` pod.

Author: Fixed.

> The default memory limit for the `argocd-redis` caching server is `256M`. Memory limit is a value that constitutes a fixed amount of memory that can be allocated across all the processes in a pod.

Reviewer 3: ```suggestion

```

Reviewer 3: I am not sure why we are giving an explanation of what memory limit is here. I does not fit here in my opinion. If we want to provide it somewhere, then maybe in the previous section?

Also, we do not need to state the default limit for Redis here as it is stated in the table above this section, so I believe having this information here again is not necessarry.

Author: Ok, that makes sense. I will remove this sentence. Fixed.

> In some instances, the default memory limit in the `argocd-redis` pod may not be sufficient to manage a large number of resources. Therefore, you must increase the memory limit to initialize the pod. After this value is increased, you must monitor the memory metrics of the Redis pod, and change the memory configuration as the application deployment scales up.

Author: Fixed.

> .Example command to view the memory configuration for the Redis pod.

Reviewer 3: I think making this into a sentence is slightly better, because we can properly explain what the user is about to see. WDYT?

> .Example command to view the memory configuration for the Redis pod.

Author: Yeah, I agree. I am keeping it as a sentence. Fixed.

[source,terminal]
----
oc get argocd -n openshift-gitops openshift-gitops -o json | jq '.spec.redis.resources'
----

.Example Output
[source,terminal]
----
{
    "limits": { #<1>
        "cpu": "500m",
        "memory": "256Mi"
  },
  "requests": { #<2>
    "cpu": "250m",
    "memory": "128Mi"
  }
}
----
<1> The highest resource limit threshold allocated to the pod 

> <2> The lowest resource limit threshold allocated to the pod

Author: Fixed.

> .Example command to change the memory configuration of the Redis pod.

Reviewer 3: I might suggest to provide a little bit more information, as follows:
```suggestion
The following example command changes the memory configuration for a Redis pod.  The highest resource limit threshold is set to 8 GiB and the lowest is set to 256 MiB.
```
WDYT?

Author: Fixed.

$ oc patch argocd -n openshift-gitops openshift-gitops --type json -p '[{"op": "replace", "path":  \
  "/spec/redis/resources/limits/memory", "value": "8Gi"}, {"op": "replace", "path": \

>   "/spec/redis/resources/requests/memory", "value": "256Mi"}]' \

Reviewer 4: ```suggestion
  "/spec/redis/resources/requests/memory", "value": "256Mi"}]'
```

Reviewer 4: This will fix the build
